wrapper/src/main/java/software/amazon/jdbc/Driver.java

```diff
@@ -139,13 +158,30 @@ public Connection connect(final String url, final Properties info) throws SQLExc
       PARENT_LOGGER.setLevel(logLevel);
     }
 
-    final TargetDriverDialectManager targetDriverDialectManager = new TargetDriverDialectManager();
-    final TargetDriverDialect targetDriverDialect =
-        targetDriverDialectManager.getDialect(driver, info);
+    TargetDriverDialect targetDriverDialect = configurationProfile == null
+        ? null
+        : configurationProfile.getTargetDriverDialect();
 
-    final ConnectionProvider connectionProvider = new DriverConnectionProvider(driver, targetDriverDialect);
+    if (targetDriverDialect == null) {
+      final TargetDriverDialectManager targetDriverDialectManager = new TargetDriverDialectManager();
+      targetDriverDialect = targetDriverDialectManager.getDialect(driver, props);
+    }
+
+    final ConnectionProvider defaultConnectionProvider =
+        new DriverConnectionProvider(driver);
+
+    ConnectionProvider effectiveConnectionProvider = null;
+    if (configurationProfile != null) {
+      effectiveConnectionProvider = configurationProfile.getConnectionProvider();
+    }
 
-    return new ConnectionWrapper(info, driverUrl, connectionProvider);
+    return new ConnectionWrapper(
```

Comment:
is it needed here to pass along both the default and the effective connection provider?